resources/user.py

```diff
@@ -167,7 +167,7 @@ def post(self):
         if user and user.archived:
             return {"message": "Invalid user"}, 403
 
-        if user.check_pw(data['password']):
+        if user and user.check_pw(data['password']):
```

Comment:
I'm not sure if this null check is needed here. Could we get a test added for these null checks?

Reply:
I checked and this gives a more accurate response from the backend. From a 500 to a 401. Just try to log in with a non-existent user and it fixes the 500 (member access on a None).

Reply:
Great! Can we get a test added?